jmilic1/code-cadets-2021
# Homework 3

To test homework 
1) run script in lecture_3/03_project/script.sh to initialize the database and send `betsRecieved` to queue.
2) run main program in controller folder and calculator folder by setting up a run configuration for both and setting their respective .env files.
3) wait for programs to populate their repositories
4) run eventsettler/main.go to send `eventSettled` messages to queue.

After running all of this I tested the application by reading console of `eventSettler/main.go`, reading the ids in console (which are selectionIds in database) and querying `SELECT * FROM bets WHERE selectionId == <id>` from controller database which is located in db/bets_db.

If console says that a certain Id has 'lost' status, all of the bets that satisfy query should have status 'lost' and payout 0.
If console says Id has 'won' status, all of the bets should have status 'won' and a non-zero payout.
